### PR TITLE
Add OTEL to taxes app

### PR DESCRIPTION
**Pull request by lkostrowski**
## Scope of the PR

- extracted shared logger based on tslog pacage
- transports for sentry and otel
- otel implemented in Taxes app

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

**Comment by lkostrowski** [comment]
@saleor/delivery-engineering-js fyi

File: apps/taxes/.env.example [hunk 1]
```diff
@@ -16,4 +16,9 @@ ALLOWED_DOMAIN=
 # If developed with tunnels, set this empty, it will fallback to default Next's localhost:3000
 # https://docs.saleor.io/docs/3.x/developer/extending/apps/local-app-development
 APP_IFRAME_BASE_URL=
-APP_API_BASE_URL=_PATTERN=
\ No newline at end of file
+APP_API_BASE_URL=_PATTERN=
```
Comment (peelar) [inline apps/taxes/.env.example:19]: is everything okay with this var?

**Review by peelar — APPROVED** [review]

File: apps/taxes/.env.example (hunk 1 above)
Comment (lkostrowski) [reply apps/taxes/.env.example:19]: good catch

File: apps/cms-v2/src/logger.ts [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
 
-/*
- * TODO: Add loggerContext
- * TODO: Migrate to shared
- */
-export const logger = new Logger<ILogObj>({
-  argumentsArrayName: "payload",
-  minLevel: parseInt(process.env.NEW_APP_LOG_LEVEL ?? "3", 10),
-});
+logger.settings.maskValuesOfKeys = ["metadata", "username", "password", "apiKey"];
```
Comment (witoszekdev) [inline apps/cms-v2/src/logger.ts:3]: question: If this logger is imported from a shared package, won't that cause interference that we change a global value?

**Review by witoszekdev — APPROVED** [review]

File: apps/cms-v2/src/logger.ts (hunk 2 above)
Comment (lkostrowski) [reply apps/cms-v2/src/logger.ts:3]: each app is built separately so this code is never executed for other apps

**Pull request merged**